### PR TITLE
Fix #1888 Convert toasts to snackbars

### LINKED ISSUE #1888: Converting toasts to snackbars.
In a lot of places in the application, we are using toasts. Unless we need a message to be displayed even after an activity is destroyed, we should use snackbars. 

Benefits of snackbars over toasts:
- More customizable than a toast
- In certain places, due to inappropriate permissions, a toast is shown. Here instead, a snack bar with an action can be shown. This will enhance the UX of the application immensely. 
- Can be swiped off the screen
- Look better than toasts, hence make for a good UI along with good UX. 

Only in places where a message needs to be displayed even when the activity is destroyed, or in the background, we should continue using toasts. For example, when the server in host books is stopped using the notification, the user might not even be in the app and might be on the phone home screen. In a case like this, a toast should be shown. 

**Additional context**

If this is an appropriate issue, I would like to work on it.

### PR DESCRIPTION
**Pull request by yashk2000**
Fixes #1888 

The toasts mention in #1888 have been converted into snackbars. 

**Screenshots** 

screencap

screencap

screencap

screencap

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt [hunk 1]
```diff
@@ -73,7 +73,7 @@ class KiwixMainActivity : CoreMainActivity() {
       val filePath = FileUtils.getLocalFilePathByUri(applicationContext, data)
 
       if (filePath == null || !File(filePath).exists()) {
-        toast(R.string.error_file_not_found)
+        Snackbar.make(currentWebView, R.string.error_file_not_found, Snackbar.LENGTH_SHORT).show()
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/main/KiwixMainActivity.kt]: there is a snackbar extension on `View`, use `currentWebView.snack`

File: core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java [hunk 2]
```diff
@@ -206,7 +207,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
         allBookmarks.clear();
         bookmarksList.clear();
         bookmarksAdapter.notifyDataSetChanged();
-        Toast.makeText(this, R.string.all_bookmarks_cleared_toast, Toast.LENGTH_SHORT).show();
+        Snackbar.make(getWindow().getDecorView().getRootView(), R.string.all_bookmarks_cleared, Snackbar.LENGTH_SHORT).show();
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java]: you couldn't use `getView()`?

Comment (yashk2000) [reply core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java]: I tried that. It didn't work, so I had to use this.

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt [hunk 3]
```diff
@@ -142,6 +145,20 @@ class LibraryFragment : BaseFragment() {
     }
   }
 
+  private fun noInternetSnackbar() {
+    val snackbar =
+      Snackbar.make(view!!.rootView, R.string.no_network_connection, Snackbar.LENGTH_LONG)
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt]: this should use the `snack` extension method too

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt [hunk 4]
```diff
@@ -142,6 +145,20 @@ class LibraryFragment : BaseFragment() {
...
+    snackbar.setAction(R.string.settings, View.OnClickListener {
+      openNetworkSettings()
+    })
+    snackbar.show()
+  }
+
+  private fun openNetworkSettings() {
+    val intent = Intent(Settings.ACTION_WIFI_SETTINGS)
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt]: inline this variable, the name doesn't add anything

File: core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java (hunk 2 above)
Comment (macgills) [reply core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java]: Use any of the views provided by butterknife, I had a brainfart, Activities do not have getView.

The way snackbar works is it traverses up the view heirarchy for the nearest CoordinatorLayout and then plops the snackbar into it, hopefully moving views with the right behaviours, ie FABs.

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt [hunk 5]
```diff
@@ -142,6 +144,18 @@ class LibraryFragment : BaseFragment() {
     }
   }
 
+  private fun noInternetSnackbar() {
+    view?.snack(
+      R.string.no_network_connection,
+      R.string.settings,
+      { (::openNetworkSettings)(Intent(Settings.ACTION_WIFI_SETTINGS)) }
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt]: what is happening here? you should be able to just say `::openNetworkSettings` without the `{}` or anything else.

Ahhhhh you are invoking a method reference and passing an intent as a parameter, yeah still no.

File: app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt [hunk 6]
```diff
@@ -142,6 +144,18 @@ class LibraryFragment : BaseFragment() {
...
+    )
+  }
+
+  private fun openNetworkSettings(intent: Intent) {
+    startActivity(intent)
```
Comment (macgills) [inline app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt]: This should just be `startActivity(Intent(Settings.ACTION_WIFI_SETTINGS))`

Comment (yashk2000) [reply app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt]: I'll update it. Sorry, I misinterpreted the previous comment.

Comment (yashk2000) [reply app/src/main/java/org/kiwix/kiwixmobile/zim_manager/library_view/LibraryFragment.kt]: @macgills updated.

**Pull request merged**